.github/actions/setup/action.yml

```diff
@@ -20,25 +20,28 @@ runs:
       with:
         version: nightly
 
-    # Workaround for parallel building with forge:
-    # https://github.com/foundry-rs/foundry/issues/4736
-    - name: Install solc
-      shell: bash
-      run: cargo install svm-rs && svm install 0.8.13
-
     - name: Setup protoc
       uses: arduino/setup-protoc@v1
 
     - name: Install node dependencies
       shell: bash
       run: pnpm install --frozen-lockfile
 
+    # Workaround for parallel building with forge:
+    # https://github.com/foundry-rs/foundry/issues/4736
+    - name: Setup forge
+      shell: bash
+      working-directory: packages/schema-type
+      run: forge build
+
     - name: Cache turbo build
       uses: actions/cache@v3
       with:
         path: .turbo
-        key: ${{ runner.os }}-turbo-${{ github.sha }}
+        key: ${{ runner.os }}-turbo-${{ github.ref_name }}-${{ github.sha }}
         restore-keys: |
+          ${{ runner.os }}-turbo-${{ github.ref_name }}-
+          ${{ runner.os }}-turbo-main
```

Comment:
In the case of branch of a branch, maybe removing `${{ runner.os }}-turbo-main` and just matching to the most recent `${{ runner.os }}-turbo-` might be better, as the CI/CD is becoming very fast and likely the devs approving these ci/cd processes will be focusing on individual problems at a time.